### PR TITLE
Implement scrollable parking spot list with clever filtering.

### PR DESCRIPTION
**Pull request by AlexandreBeeDeeBop**
Create a scrollable list of parking spots in the 'SpotListScreen' composable, displaying detailed information on each card. The `SpotCard` composable shows parking spot details,
 while the `FilterHeader` allows users to apply advanced
filters to sort spots based on multiple criteria such as protection, rack type, and capacity,
showing the most relevant results first.

### CODE REVIEW

File: app/src/main/java/com/github/se/cyrcle/ui/list/List.kt [hunk 1]
```diff
@@ -0,0 +1,449 @@
...
+import com.github.se.cyrcle.model.parking.*
+import com.github.se.cyrcle.ui.theme.DarkBlue
+import kotlin.math.atan2
+import kotlin.math.cos
+import kotlin.math.sin
+import kotlin.math.sqrt
+
+// EXAMPLES
+
+val referencePoint1 =
```
Comment (LouisJamesVasseur) [inline app/src/main/java/com/github/se/cyrcle/ui/list/List.kt:26]: These could be left for the test files

File: app/src/main/java/com/github/se/cyrcle/ui/list/List.kt [hunk 2]
```diff
@@ -0,0 +1,449 @@
...
+            }
+      }
+}
+
+// Extension fuction to toggle selection
+fun <T> Set<T>.toggle(item: T): Set<T> {
+  return if (this.contains(item)) this - item else this + item
+}
+
+fun calculateDistance(point1: Point, point2: Point): Double {
```
Comment (LouisJamesVasseur) [inline app/src/main/java/com/github/se/cyrcle/ui/list/List.kt:435]: We'll need to modify this to include viewModel-based distance modifications, but this works just fine for now

**Review by LouisJamesVasseur — APPROVED** [review]
This looks all good. We'll need to get back to this to increase code coverage and connect the screen to the ViewModel. We'll also need to increase code coverage and add more tests

**Review by Pijukai — APPROVED** [review]
Good to go well done !

**Pull request merged**